cli/pkg/artifact_builder/artifact_builder.go

```diff
@@ -384,9 +385,106 @@ func (ab ArtifactBuilder) push(ctx context.Context, tags []string, servicesImage
 			}
 		}
 	}
+
+	ab.cveScan(ctx, serviceRegistries, servicesImage, tags)
+
 	return nil
 }
 
+func (ab ArtifactBuilder) cveScan(ctx context.Context, serviceRegistries map[string]*config.RegistryConfig, servicesImage map[string]string, tags []string) {
+	log.Info("Scanning images for vulnerabilities...")
+	ecrClient := ab.backend.GetECRClient()
+	vulnerabilityCount := 0
+
+	for serviceName, registry := range serviceRegistries {
+		if _, ok := servicesImage[serviceName]; !ok {
+			continue
+		}
+
+		for _, currentTag := range tags {
+			result, descriptor, err := ab.getRegistryImages(ctx, registry, currentTag)
+			if err != nil {
+				log.Errorf("error getting Image: %s", err.Error())
+				continue
+			}
+
+			// TODO: Repository scan on push is deprecated, allow registry scan on push configuration to be considered as well
+			out, err := ecrClient.BatchGetRepositoryScanningConfiguration(ctx, &ecr.BatchGetRepositoryScanningConfigurationInput{
+				RepositoryNames: []string{descriptor.RepositoryName},
+			})
+
+			if err != nil {
+				log.Errorf("error getting repository scanning configuration: %s", err.Error())
+				continue
+			}
+
+			if len(out.ScanningConfigurations) == 0 {
+				continue
+			}
+
+			if !out.ScanningConfigurations[0].ScanOnPush {
```

Comment:
and this is set by the stack configuration?

Reply:
Yes, this is set by `scan_on_push = "true"` on a service. AWS deprecated the scan on push for individual repositories, and added a global registry setting, so this PR will have a followup.